### ISSUE TITLE
update_cached_attributes makes multiple requests to external resource

### ISSUE THREAD

**Issue by arielvalentin** [innovationfactory/has_remote#1]
# While testing I noticed that the has_remote sends multiple requests to the external resource for each remote attribute that I specified for synchronizing with local_cache:

```
    self.class.cached_attributes.each do |remote_attr|
      local_attr = self.class.remote_attribute_aliases[remote_attr] || remote_attr
      write_attribute(local_attr, has_remote? ? remote(true).send(remote_attr) : nil)
    end
```
# 

For each of the attributes 
1. has_remote? # invokes remote(true), which sends a request to !remote(true).nil? 
2. remote(true).send #sends a request to fetch the resource 

In order to reduce the number of requests I moved the remote(true) call to a local variable outside of the loop and allowed for the ActiveResource errors to bubble up to the invoker. Do you think this is reasonable to do?

**Comment by thoughtless** [comment]
This definitely needs to be made more efficient. If a fix can't be made immediately, how about at least a warning in the Rdocs?

**Comment by s-andringa** [comment]
Sorry it took so long but a patch has now been released in version 0.1.7 (see https://github.com/innovationfactory/has_remote/compare/912da57...8501b9#L0L156 for changes.)
